### PR TITLE
Samples include UserName and Password validation, and Fix in CoreWCF to allow it to work 

### PR DESCRIPTION
**Pull request by JohnLeyva**
First commit has a fix to allow UserNameCredentialsValidation to work.

Samples UserName and Password validation

Common URLs in a shared Settings project that facilitates keeping the paths used in the sample projects

### CODE REVIEW

File: src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs [hunk 1]
```diff
@@ -488,7 +488,7 @@ public Task OpenAsync()
 
         public Task OpenAsync(CancellationToken token)
         {
-            throw new NotImplementedException();
+            return Task.CompletedTask;
```
Comment (birojnayak) [inline src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs:491]: looks good.. bdw this never gets called, did you hit a situation where this was getting called ?

**Review by birojnayak — COMMENTED** [review]
looks ok to me... @mconnew any concern ?

File: src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs (hunk 1 above)
Comment (JohnLeyva) [reply src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs:491]: Yes, Indeed it was called consistenly from the sample application and a few times tests also called it, but the behaviour was not deterministic.

Wondering if it is related to having the debugger attached.

After, trying to follow it, in the end, I decided to implement the method. But still curios about what is really happening there.

File: CustomUserNamePasswordValidator.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System.ServiceModel;
+
+namespace DesktopServer
+{
+    internal class CustomUserNamePasswordValidator : System.IdentityModel.Selectors.UserNamePasswordValidator
```
Comment (mconnew) [inline CustomUserNamePasswordValidator.cs:5]: This file and the Core file need to be in the relevant project folders and not at the root.

File: src/Samples/DesktopServer/Program.cs [hunk 3]
```diff
@@ -1,25 +1,42 @@
...
-            host.AddServiceEndpoint(contract, new WSHttpBinding(SecurityMode.None), "/wsHttp.svc");
-            host.AddServiceEndpoint(contract, new WSHttpBinding(SecurityMode.Transport), "/wsHttp.svc");
+            string hostname = "localhost"; //"localtest"
+
+            IEnumerable<string> baseAddressList = new[] {
+                "net.tcp://localhost:8089/",
+                "http://localhost:8088/",
+                "https://localhost:8443/" };
+            baseAddressList = baseAddressList.Select(i => i.Replace("//localhost:", $"//{hostname}:"));
+            Uri[] baseUriList = baseAddressList.Select(a => new Uri(a)).ToArray();
```
Comment (mconnew) [inline src/Samples/DesktopServer/Program.cs:22]: This replacing of the hostname isn't needed as you added an optional parameter to SetDefaults to do this for you.

File: src/Samples/StandardCommon/Settings.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        public bool UseHttps { get; set; } = true;
+        public Uri basicHttpAddress { get; set; }
+        public Uri basicHttpsAddress { get; set; }
+        public Uri wsHttpAddress { get; set; }
+        public Uri wsHttpAddressValidateUserPassword { get; set; }
+        public Uri wsHttpsAddress { get; set; }
+        public Uri netTcpAddress { get; set; }
+
+        public Settings SetDetaults(string hostname = "localhost")
+        {
```
Comment (mconnew) [inline src/Samples/StandardCommon/Settings.cs:16]: SetDefaults

File: src/Samples/gencert.ps1 [hunk 5]
```diff
@@ -40,8 +40,9 @@ function Set-SSL_WCF_ServiceHost {
     $certHash = "certhash=$($certificateObject.Thumbprint)"
     $appid = "appid={1d7985f5-e1b0-4fd4-bdeb-7e28590783da}"
 
-#    netsh http add sslcert ipport=127.0.0.1:8443 $certHash $appid certstorename=Root
+    netsh http add sslcert ipport=127.0.0.1:8443 $certHash $appid certstorename=Root
     netsh http add sslcert hostnameport=localhost:8443 $certHash $appid certstorename=Root
+    netsh http add sslcert hostnameport=localtest:8443 $certHash $appid certstorename=Root
 #    Netsh http delete sslcert ipport=127.0.0.1:8443
```
Comment (mconnew) [inline src/Samples/gencert.ps1:46]: Do we need localtest in here? Looks like it's specific to your dev environment. There's also a little bit of remnants of this still in some of the test code.

Comment (JohnLeyva) [reply src/Samples/gencert.ps1:46]: I'll remove it. 
I added it in order to be able to use fiddler.

**Comment by JohnLeyva** [comment]
@mconnew, @birojnayak, please let me know if there is something pending

**Comment by mconnew** [comment]
@JohnLeyva, thanks for making the requested changes. Merging now.

**Pull request merged**